src/test/testAccounts.ts

```diff
@@ -0,0 +1,159 @@
+import { ContractTransaction, ethers } from "ethers";
+import { getConfig } from "../config";
+import { register, Registration } from "../core/contracts/registry";
+import { Identity__factory, Registry__factory } from "../types/typechain";
+
+export interface RegistrationWithSigner extends Registration {
+  signer: ethers.Signer;
+}
+
+/**
+ * Use this function to set up a new signer other than what is in the config.
+ * @param accountIndex the index in the TestAccounts to get the signer for.
+ * @return the ethers.Signer associated with the test account
+ */
+export const getSignerForAccount = (accountIndex: number): ethers.Signer => {
+  if (accountIndex >= TestAccounts.length) throw new Error(`there are only ${TestAccounts.length} accounts.`);
+  const { provider } = getConfig();
+  if (!provider) throw new Error("no provider configured");
+  return new ethers.Wallet(TestAccounts[accountIndex].privateKey, provider);
+};
+
+/**
+ * Parses a DSNP Id in a contract transaction log event.
+ * @param transaction
+ * @returns the DSNP Id
+ */
+export const getIdFromRegisterTransaction = async (transaction: ContractTransaction): Promise<any> => {
+  const receipt = await transaction.wait(1);
+  const reg = Registry__factory.createInterface();
+  const event = reg.parseLog(receipt.logs[0]);
+  return event.args[0];
+};
+
+/**
+ * Creates a new DSNP Identity Proxy contract using the specified test account, and registers the
+ * provided handle. Callers must keep track of what accounts have already been used.
+ * @param acctIdx the index in TestAccounts to use.
+ * @param handle the handle to register
+ * @return a RegistrationWithSigner object
+ */
+export const newRegistrationForAccountIndex = async (
+  acctIdx: number,
+  handle: string
+): Promise<RegistrationWithSigner> => {
+  const signer = getSignerForAccount(acctIdx);
+  const newSignerAddr = await signer.getAddress();
+  const identityContract = await new Identity__factory(signer).deploy(newSignerAddr);
+  await identityContract.deployed();
+  const contractAddr = identityContract.address;
+  const tx2 = await register(contractAddr, handle);
+  const dsnpId = await getIdFromRegisterTransaction(tx2);
+  return { contractAddr, dsnpId, handle, signer };
+};
+
+const TestAccounts = [
```

Comment:
Not blocking, but maybe we should move this to the top of the files and all-caps the variable name since it seems like a constant.